yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,256 @@
+                        'thumbnail': 'https://cdn.vidyard.com/thumbnails/5860926/JIOaJR08dM4QgXi_iQ2zGA_small.png',
+                        'duration': 191,
+                    },
+                    {
+                        'id': 'U9hEsKlpPTMasqEf5n0KQg',
+                        'display_id': '41976382',
+                        'ext': 'mp4',
+                        'title': 'Adjust the Panels for Palm Beach Polysatin Shutters With BiFold Track',
+                        'thumbnail': 'https://cdn.vidyard.com/thumbnails/5860687/CwHxBv4UudAhOh43FVB4tw_small.png',
+                        'duration': 138,
+                    },
+                    {
+                        'id': 'gBdBzCvGXK5j2f20lAF-vg',
+                        'display_id': '41976409',
+                        'ext': 'mp4',
+                        'title': 'Assemble and Install the Valance for Palm Beach Polysatin Shutters With BiFold Track',
+                        'thumbnail': 'https://cdn.vidyard.com/thumbnails/5861425/0y68qlMU4O5VKU7bJ8i_AA_small.png',
+                        'duration': 148,
+                    },
+                ],
+            },
+            'playlist_count': 6,
+        },
+        {
+            # URL of iframe embed src
+            'url': 'https://play.vidyard.com/iDqTwWGrd36vaLuaCY3nTs.html',
+            'info_dict': {
+                'id': 'lrYeWCqR3UCAzwqUAgegHg',
+                'display_id': '9281009',
+                'ext': 'mp4',
+                'title': 'Lightbox Embed',
+                'thumbnail': 'https://cdn.vidyard.com/thumbnails/spacer.gif',
+                'duration': 39,
+            },
+        },
+    ]
+    _EMBED_REGEX = [
+        # iframe embed
+        r'<iframe[^>]+?src=(["\'])(?P<url>(?:https?:)?//play\.vidyard\.com/[\w-]+.\w+)\1',
+    ]
+    _WEBPAGE_TESTS = [
+        {
+            # URL containing inline/lightbox embedded video
+            'url': 'https://resources.altium.com/p/2-the-extreme-importance-of-pc-board-stack-up',
+            'info_dict': {
+                'id': 'rGGKQEAu8X-APkU68K8U_w',
+                'display_id': '3225198',
+                'ext': 'mp4',
+                'title': 'The Extreme Importance of PC Board Stack Up',
+                'thumbnail': 'https://cdn.vidyard.com/thumbnails/73_Q3_hBexWX7Og1sae6cg/9998fa4faec921439e2c04_small.jpg',
+                'duration': 3422,
+            },
+        },
+    ]
+
+    @classmethod
+    def _extract_embed_urls(cls, url, webpage):
+        # Handle protocol-less embed URLs
+        for embed_url in super()._extract_embed_urls(url, webpage):
+            if embed_url.startswith('//'):
+                embed_url = f'https:{embed_url}'
+            yield embed_url
+
+        # Extract inline/lightbox embeds
+        for embed_elm in re.findall(r'(<img[^>]+class=(["\'])(?:[^>"\']* )?vidyard-player-embed(?: [^>"\']*)?\2[^>]+[^>]*>)', webpage):
+            embed = extract_attributes(embed_elm[0]) or {}
```

Comment:
I believe `extract_attributes()` always returns a `dict`, even if empty.
```suggestion
            embed = extract_attributes(embed_elm[0])
```